waiter/src/waiter/service_description.clj

```diff
@@ -1075,8 +1076,8 @@
   "Returns whether the `username` is allowed to modify the specified service description."
   [kv-store entitlement-manager service-id username]
   ; the stored service description should already have a run-as-user
-  (let [service-description (service-id->service-description kv-store service-id {} [])]
-    (authz/manage-service? entitlement-manager username service-id service-description)))
+  (let [core-service-description (service-id->service-description kv-store service-id {} [])]
```

Comment:
This code actually just wants the run-as-user. It should get it directly via fetch-core kv-store service-id

Reply:
Done, thanks for noticing this.